lib/hackney/income/tenancy_prioritiser/tenancy_classification.rb

```diff
@@ -113,13 +100,16 @@ def send_warning_letter?
 
         def send_nosp?
           valid_actions = [
-            Hackney::Tenancy::ActionCodes::PRE_NOSP_WARNING_LETTER_SENT
+            Hackney::Tenancy::ActionCodes::INCOME_COLLECTION_LETTER_2,
+            Hackney::Tenancy::ActionCodes::INCOME_COLLECTION_LETTER_2_UH
           ]
 
           can_send_nosp = false
 
           if @criteria.nosp_expiry_date.present?
             can_send_nosp = @criteria.nosp_expiry_date < Time.zone.now
+          elsif @criteria.active_agreement?
+            can_send_nosp = false
```

Comment:
Could we not just do:

```suggestion
            return false
```

Reply:
Indeed, I've changed this!

Reply:
Fixed this in 34a7f84ed